src/app/components/elements/views/ActivityGraph.tsx

```diff
@@ -1,17 +1,24 @@
-import React, { useEffect } from "react";
-import { bb } from "billboard.js";
+import React, { useEffect, useMemo } from "react";
+import bb, { zoom, areaSpline } from "billboard.js";
 import { AnsweredQuestionsByDate } from "../../../../IsaacApiTypes";
 import { formatISODateOnly } from "../DateString";
 
-export const ActivityGraph = ({ answeredQuestionsByDate }: { answeredQuestionsByDate: AnsweredQuestionsByDate }) => {
-  let selectedDates: string[] = [];
+const filterSelectedDates = (answeredQuestionsByDate: AnsweredQuestionsByDate) => {
   const foundDates = answeredQuestionsByDate ? Object.keys(answeredQuestionsByDate) : [];
   if (foundDates && foundDates.length > 0) {
```

Comment:
I think you may be able to simplify this code even further, because both `foundDates` is always an array, so the result of `foundDates.filter` will always work, even when `foundDates` is empty (in that case `filter` won't run and the result (`nonZeroDates`) will be an empty array).